node/src/components/block_validator.rs

```diff
@@ -141,40 +171,27 @@ where
                         let fetch_effects: Effects<Event<T, I>> = block_deploys
                             .iter()
                             .flat_map(|deploy_hash| {
-                                let chainspec = Arc::clone(&chainspec);
                                 // For every request, increase the number of in-flight...
                                 in_flight.inc(deploy_hash);
-
                                 // ...then request it.
-                                let deploy_hash = *deploy_hash;
-                                let validate_deploy =
-                                    move |result: FetchResult<Deploy, I>| match result {
-                                        FetchResult::FromStorage(deploy)
-                                        | FetchResult::FromPeer(deploy, _) => {
-                                            if deploy
-                                                .header()
-                                                .is_valid(&chainspec.deploy_config, block_timestamp)
-                                            {
-                                                Event::DeployFound(deploy_hash)
-                                            } else {
-                                                info!(%deploy_hash, "deploy is invalid");
-                                                Event::DeployMissing(deploy_hash)
-                                            }
-                                        }
-                                    };
-                                effect_builder
-                                    .fetch_deploy(deploy_hash, sender.clone())
-                                    .map_or_else(validate_deploy, move || {
-                                        info!(%deploy_hash, "deploy missing");
-                                        Event::DeployMissing(deploy_hash)
-                                    })
+                                fetch_deploy(
+                                    effect_builder,
+                                    Arc::clone(&chainspec),
+                                    block_timestamp,
+                                    *deploy_hash,
+                                    sender.clone(),
+                                )
                             })
                             .collect();
                         effects.extend(fetch_effects);
 
                         entry.insert(BlockValidationState {
                             missing_deploys,
                             responders: smallvec![responder],
+                            sources: VecDeque::new(), /* This is empty b/c we create the first
+                                                       * request
+                                                       * using `sender`. */
```

Comment:
```suggestion
                            sources: VecDeque::new(), /* This is empty b/c we create the first
                                                       * request using `sender`. */
```
Or maybe even better:
```suggestion
                            // This is empty b/c we create the first request using `sender`.
                            sources: VecDeque::new(),
```

Reply:
That's what `cargo fmt` does, not me.